devbox.go

```diff
@@ -266,3 +295,79 @@ func findConfigDir(dir string) (string, error) {
 	}
 	return "", missingDevboxJSONError(dir)
 }
+
+// installMode is an enum for helping with ensurePackagesAreInstalled implementation
+type installMode string
+
+const (
+	install   installMode = "install"
+	uninstall installMode = "uninstall"
+)
+
+func (d *Devbox) ensurePackagesAreInstalled(mode installMode) error {
+	if err := d.Generate(); err != nil {
+		return err
+	}
+
+	installingVerb := "Installing"
+	if mode == uninstall {
+		installingVerb = "Uninstalling"
+	}
+	fmt.Fprintf(d.writer, "%s nix packages. This may take a while...", installingVerb)
+
+	// We need to re-install the packages
+	if err := d.ApplyDevNixDerivation(); err != nil {
+		fmt.Println()
+		return err
+	}
+	fmt.Println("done.")
+
+	return nil
+}
+
+func (d *Devbox) printPackageUpdateMessage(mode installMode, pkgs []string) error {
+	// (Only when in devbox shell) Prompt the user to run `hash -r` to ensure their
+	// shell can access the most recently installed binaries, or ensure their
+	// recently uninstalled binaries are not accidentally still available.
+	if len(pkgs) > 0 && IsDevboxShellEnabled() {
+		installedVerb := "installed"
+		if mode == uninstall {
+			installedVerb = "removed"
+		}
+
+		successMsg := fmt.Sprintf("%s is now %s.", pkgs[0], installedVerb)
```

Comment:
Just say "packages updated?"

Reply:
I had considered that, but "updated package" has another meaning i.e. <pkg> has been updated from version 1.0.1 to 1.0.2 so felt it better to be super explicit

Reply:
oh right, haven't thought about that hehe